ArmPlatformPkg/Sec/Sec.c

```diff
@@ -107,14 +116,15 @@ CreatePpiList (
   ListBase = PcdGet64 (PcdCPUCoresStackBase);
   CopyMem ((VOID *)ListBase, gCommonPpiTable, sizeof (gCommonPpiTable));
   CopyMem ((VOID *)(ListBase + sizeof (gCommonPpiTable)), PlatformPpiList, PlatformPpiListSize);
-
-  // Set the Terminate flag on the last PPI entry
-  LastPpi = (EFI_PEI_PPI_DESCRIPTOR *)ListBase +
-            ((sizeof (gCommonPpiTable) + PlatformPpiListSize) / sizeof (EFI_PEI_PPI_DESCRIPTOR)) - 1;
-  LastPpi->Flags |= EFI_PEI_PPI_DESCRIPTOR_TERMINATE_LIST;
+  // Copy the Transfer List PPI after the Common and Platform PPIs
+  CopyMem (
+    (VOID *)(ListBase + sizeof (gCommonPpiTable) + PlatformPpiListSize),
+    &mTransferListBase,
+    sizeof (mTransferListBase)
+    );
 
   *PpiList     = (EFI_PEI_PPI_DESCRIPTOR *)ListBase;
-  *PpiListSize = sizeof (gCommonPpiTable) + PlatformPpiListSize;
+  *PpiListSize = sizeof (gCommonPpiTable) + PlatformPpiListSize + sizeof (mTransferListBase);

```

Comment:
Take care to only add `sizeof(EFI_PEI_PPI_DESCRIPTOR)` here if `TransferListBase != NULL`